### PR TITLE
'PR-Initial'

### PR DESCRIPTION
**Pull request by MehranSiddiqui**
- Created Authentication for Google Login and Facebook Login using npm packages.
- Created a Signup Page using which we can adds a new user to Storage also checks if there is an existing user with same email id , if existing the alerts about existence.
- Created a Login Page which checks the local storage for same email id and login if the email and password are correct
- The validation was done initially manually by formik validation later upgraded to Yup validation

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,38 @@
...
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <Routes>
+        <Route
+          path="/"
+          element={
+            <>
```
Comment (kumarparth380) [inline src/App.js]: there is no need of fragment if you have only one element

File: src/Auths/Facebook-Auth/FacebookAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from "react";
+import FacebookLogin from "react-facebook-login";
+
+const FacebookAuth = () => {
+  const authId = "704948947579271";
```
Comment (kumarparth380) [inline src/Auths/Facebook-Auth/FacebookAuth.jsx]: put this in env file

File: src/Auths/Google-Auth/GoogleoAuth.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+
+import { GoogleLogin } from "react-google-login";
+
+const GoogleoAuth = () => {
+  const clientID =
+    "857942524114-ari6vree85d1udmtfrla5hlebb1ses70.apps.googleusercontent.com";
```
Comment (kumarparth380) [inline src/Auths/Google-Auth/GoogleoAuth.jsx]: save this in the env file

File: src/pages/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+import { Button } from "@mui/material";
+
+const Signup = () => {
+  const navigateTo = useNavigate();
+  const userData = localStorage.getItem("Users")
+    ? JSON.parse(localStorage.getItem("Users"))
+    : [];
+    
+    // Manual Validation
+  // const validate = (values) => {
```
Comment (kumarparth380) [inline src/pages/Signup.jsx]: remove unnecessary code

**Review by kumarparth380 — CHANGES_REQUESTED** [review]
everything looks good except for some minor changes

File: src/Auths/Facebook-Auth/FacebookAuth.jsx (hunk 2 above)
Comment (MehranSiddiqui) [reply src/Auths/Facebook-Auth/FacebookAuth.jsx]: Added an .env file as asked

File: src/Auths/Google-Auth/GoogleoAuth.jsx (hunk 3 above)
Comment (MehranSiddiqui) [reply src/Auths/Google-Auth/GoogleoAuth.jsx]: added .env as asked

File: src/pages/Signup.jsx (hunk 4 above)
Comment (MehranSiddiqui) [reply src/pages/Signup.jsx]: Removed unnecessary code

**Review by MehranSiddiqui — COMMENTED** [review]
- Added .env file which holds the API keys
- removed unnecessary code.

File: src/Auths/Facebook-Auth/FacebookAuth.jsx [hunk 5]
```diff
@@ -2,15 +2,17 @@ import React, { useState } from "react";
 import FacebookLogin from "react-facebook-login";
 
 const FacebookAuth = () => {
-  const authId = "704948947579271";
+  const authId = process.env.REACT_APP_FACEBOOK_APP_ID;
   const [user, setUser] = useState({});
   const [login, setLogin] = useState(false);
-
+console.log(authId);
```
Comment (sheelu-nickelfox) [inline src/Auths/Facebook-Auth/FacebookAuth.jsx]: remove consoles once done with logic.

File: src/Auths/Google-Auth/GoogleoAuth.jsx [hunk 6]
```diff
@@ -3,25 +3,31 @@ import React, { useState } from "react";
...
-        setUser(res.profileObj.name);
-        setLogin(true);
-      }else{
-        setLogin(false);
-      }
-  }
-  return <div>
-  
-      {!login && <GoogleLogin clientId={clientID} buttonText="Login" onSuccess={requestResponse} onFailure={requestResponse} />}
+  const requestResponse = (res) => {
```
Comment (sheelu-nickelfox) [inline src/Auths/Google-Auth/GoogleoAuth.jsx:14]: Add short comment to brief why this function, what it does.
make habit of writing short aritculative comments to explain important functions in your program.

File: src/App.js (hunk 1 above)
Comment (MehranSiddiqui) [reply src/App.js:13]: Removed Fragments as asked

File: src/Auths/Google-Auth/GoogleoAuth.jsx (hunk 6 above)
Comment (MehranSiddiqui) [reply src/Auths/Google-Auth/GoogleoAuth.jsx:14]: Okay. Sheelu will start doing that.

File: src/Auths/Facebook-Auth/FacebookAuth.jsx (hunk 5 above)
Comment (MehranSiddiqui) [reply src/Auths/Facebook-Auth/FacebookAuth.jsx]: Removed

File: src/Auths/Google-Auth/GoogleoAuth.jsx (hunk 6 above)
Comment (MehranSiddiqui) [reply src/Auths/Google-Auth/GoogleoAuth.jsx:14]: Added short comments to give brief info about the function..

**Pull request merged**